actors/miner/src/lib.rs

```diff
@@ -1153,22 +1154,17 @@ impl Actor {
             update_sector_infos.push(UpdateAndSectorInfo { update, sector_info });
         }
 
-        let activation_infos: Vec<DealsActivationInfo> = update_sector_infos
-            .iter()
-            .map(|usi| DealsActivationInfo {
-                deal_ids: usi.update.deals.clone(),
-                sector_number: usi.update.sector_number,
-                sector_expiry: usi.sector_info.expiration,
-                sector_type: usi.sector_info.seal_proof,
-            })
-            .collect();
-        // Request CommD computation while activating deals.
-        let (batch_return, deals_spaces) =
-            batch_activate_deals_and_claim_allocations(rt, &activation_infos, true)?;
+        let data_activations: Vec<DealsActivationInputCheckCommD> =
+            update_sector_infos.iter().map(|x| x.clone().into()).collect();
+
+        let (batch_return, data_activations) = activate_deals_check_commd(rt, &data_activations)?;
```

Comment:
This isn't quite right - see my comment on `impl From` below. The CommD was not necessarily provided. So it must be requested but can't necessarily be checked.